Adding new options and example scenarios (#28976)
dotnet/docs: docs/core/additional-tools/dotnet-coverage.md

D:\serverexample\server>
```

### Server and client mode

> Code coverage collection can be done in server-client mode as well. In this scenario, a code coverage collection server starts. And multiple clients can connect with the server. And code coverage is collected for all the clients collectively.

Reviewer: ```suggestion
Code coverage collection can be done in server-client mode as well. In this scenario, a code coverage collection server starts, and multiple clients can connect with the server. Code coverage is collected for all the clients collectively.
```

> Code coverage server can be started by using the following command:

Reviewer: ```suggestion
Start the code coverage server using the following command:
```

```condole
dotnet-coverage collect --session-id serverdemo --server-mode
```

> Session ID was specified as `serverdemo` for ther server. A client can connect to the server using this session ID by the following command:

Reviewer: ```suggestion
In this example, the session ID was specified as `serverdemo` for the server. A client can connect to the server using this session ID using the following command:
```

```condole
dotnet-coverage connect serverdemo dotnet run
```

> Finally, session `serverdemo` and the server can be closed as follows:

Reviewer: ```suggestion
Finally, you can close the session `serverdemo` and the server using the following command:
```

```console
dotnet-coverage shutdown serverdemo
```

> Server process will create a collective code coverage report for all clients and will exit.

Reviewer: ```suggestion
The server process creates a collective code coverage report for all clients and exits.
```

Following is an example of full output on the server side:

```console
D:\serverexample\server> dotnet-coverage collect --session-id serverdemo --server-mode
SessionId: serverdemo
// Server will be in idle state and wait for connect and shutdown commands
Code coverage results: output.coverage.
D:\serverexample\server>
```

Following is an example of full output on the client side:

```console
D:\serverexample\server> dotnet-coverage connect serverdemo ConsoleApplication.exe World
Hello World!!
D:\serverexample\server> dotnet-coverage connect serverdemo WpfApplication.exe
D:\serverexample\server> dotnet-coverage shutdown serverdemo
D:\serverexample\server>
```

> Both server and client can be started in background mode as well. It will start another process in the background and will return control back to the user.

Reviewer: ```suggestion
You can also start both server and client in background mode. Another process starts in the background and returns control back to the user.
```
